editor/scene_tree_dock.cpp

```diff
@@ -1023,51 +1023,119 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 			}
 
 			List<Node *> selection = editor_selection->get_selected_node_list();
-			List<Node *>::Element *e = selection.front();
-			if (e) {
-				Node *node = e->get();
-				if (node) {
-					bool editable = EditorNode::get_singleton()->get_edited_scene()->is_editable_instance(node);
 
-					if (editable) {
-						editable_instance_remove_dialog->set_text(TTR("Disabling \"editable_instance\" will cause all properties of the node to be reverted to their default."));
-						editable_instance_remove_dialog->popup_centered();
-						break;
+			if (selection.size() > 1) {
+				// check if the selection contains both editable and non-editable nodes.
+				bool has_editables = false;
+				bool has_non_editables = false;
+				for (int i = 0; i < selection.size(); i++) {
+					Node *node = selection[i];
+					if (node) {
+						bool editable = EditorNode::get_singleton()->get_edited_scene()->is_editable_instance(node);
+						has_editables |= editable;
+						has_non_editables |= !editable;
+					}
+				}
+
+				// Fire dialog if selection does contain both, to ask user what they want to do (enable or disable).
+				if (has_non_editables && has_editables) {
+					editable_instance_enable_or_disable_dialog->set_text(TTR("Selection contains some nodes that have \"Editable Children\" enabled and some that do not."));
```

Comment:
This is only done here, so should be moved to constructor.

Reply:
also done for `placeholder_enable_or_disable_dialog`